internal/tui/runner.go

```diff
@@ -0,0 +1,293 @@
+		case "ctrl+c":
+			m.err = fmt.Errorf("cancelled by user")
+			return m, tea.Quit
+
+		case "enter":
+			if err := m.validateCurrentAnswer(); err != nil {
+				m.setCurrentError(err.Error())
+				return m, nil
+			}
+
+			// Save current answer
+			current := m.questions[m.currentQuestion]
+			m.answers[current.Name] = current.Prompt.Value()
+
+			// Move to next question or finish
+			m.currentQuestion++
+			if m.currentQuestion >= len(m.questions) {
+				m.finished = true
+				return m, tea.Quit
+			}
+
+			// Clear any previous error
+			m.setCurrentError("")
+			return m, nil
+		}
+	}
+
+	// Update current prompt
+	if !m.finished && m.currentQuestion < len(m.questions) {
+		current := m.questions[m.currentQuestion]
+		updatedPrompt, cmd := current.Prompt.Update(msg)
+		current.Prompt = updatedPrompt
+		return m, cmd
+	}
+
+	return m, nil
+}
+
+func (m *questionnaireModel) validateCurrentAnswer() error {
+	current := m.questions[m.currentQuestion]
+	if current.Validate != nil {
+		return current.Validate(current.Prompt.Value())
+	}
+	return nil
+}
+
+func (m *questionnaireModel) setCurrentError(err string) {
+	if m.currentQuestion < len(m.questions) {
+		current := m.questions[m.currentQuestion]
+		switch p := current.Prompt.(type) {
+		case interface{ SetError(string) }:
+			p.SetError(err)
+		}
+	}
+}
+
+func (m *questionnaireModel) View() string {
+	if m.finished {
+		return ""
+	}
+
+	var b strings.Builder
+
+	// Header with progress
+	progress := fmt.Sprintf("Question %d of %d", m.currentQuestion+1, len(m.questions))
+	headerStyle := lipgloss.NewStyle().
+		Bold(true).
+		Foreground(lipgloss.Color("86")).
+		BorderStyle(lipgloss.NormalBorder()).
+		BorderBottom(true).
+		Width(m.width).
+		Align(lipgloss.Center)
+
+	b.WriteString(headerStyle.Render(progress))
+	b.WriteString("\n\n")
+
+	// Current question
+	if m.currentQuestion < len(m.questions) {
+		current := m.questions[m.currentQuestion]
+		b.WriteString(current.Prompt.Render())
+	}
+
+	// Footer instructions
+	b.WriteString("\n\n")
+	instructions := "Press Enter to continue, Ctrl+C to cancel"
+	if isMultiSelect(m.questions[m.currentQuestion].Prompt) {
+		instructions = "Use ↑↓ to navigate, Space to select, Enter to continue, Ctrl+C to cancel"
+	} else if isSelect(m.questions[m.currentQuestion].Prompt) {
+		instructions = "Use ↑↓ to navigate, Enter to continue, Ctrl+C to cancel"
+	}
+
+	footerStyle := lipgloss.NewStyle().
+		Foreground(lipgloss.Color("241")).
+		Italic(true)
+	b.WriteString(footerStyle.Render(instructions))
+
+	return b.String()
+}
+
+func isMultiSelect(prompt Prompt) bool {
+	_, ok := prompt.(*MultiSelect)
+	return ok
+}
+
+func isSelect(prompt Prompt) bool {
+	_, ok := prompt.(*Select)
+	return ok
+}
+
+// Ask runs multiple questions and stores answers in the provided struct
+func Ask(questions []*Question, answers interface{}) error {
+	model := newQuestionnaireModel(questions)
+	program := tea.NewProgram(model)
+
+	finalModel, err := program.Run()
```

Comment:
With the previous implementation users can see what they have answered in the questions before, but now one can only see the current question. Would it be possible to have this behavior with this library?
<img width="343" height="114" alt="imagen" src="https://github.com/user-attachments/assets/d677c2f9-f3b1-4c79-ad9f-f9c67fafc8c0" />